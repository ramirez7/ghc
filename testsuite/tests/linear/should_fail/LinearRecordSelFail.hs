{-# LANGUAGE LinearTypes #-}

module LinearRecordSel where

-- Only one-field one-constructor records are linear
data T1 = C { x1, x2 :: Int }
data T2 = D1 { u :: Int } | D2 { v :: Int }

f :: T1 ->. Int
f = x1

g :: T2 ->. Int
g = u
