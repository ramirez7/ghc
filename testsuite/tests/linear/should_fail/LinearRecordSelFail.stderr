
LinearRecordSelFail.hs:10:5: error:
    • Nontrivial multiplicity equalities are currently not supported
    • In the expression: x1
      In an equation for ‘f’: f = x1

LinearRecordSelFail.hs:10:5: error:
    • Couldn't match type ‘'Omega’ with ‘'One’
      Expected type: T1 ->. Int
        Actual type: T1 -> Int
    • In the expression: x1
      In an equation for ‘f’: f = x1

LinearRecordSelFail.hs:13:5: error:
    • Nontrivial multiplicity equalities are currently not supported
    • In the expression: u
      In an equation for ‘g’: g = u

LinearRecordSelFail.hs:13:5: error:
    • Couldn't match type ‘'Omega’ with ‘'One’
      Expected type: T2 ->. Int
        Actual type: T2 -> Int
    • In the expression: u
      In an equation for ‘g’: g = u
